protostar/cheatable_starknet/cheaters/contracts.py

```diff
@@ -253,14 +218,28 @@ async def call(
         contract_address: Address,
         function_name: str,
         cheaters: "CairoCheaters",
+        unregister_expected_call: Callable,
```

Comment:
I don't really like passing it like this, @Arcticae is there a way I can access a cheater from another cheater? Would it make sense to create such links? Or passing stuff like this is alright?
Right now, I access this from the level of the cheat code, via the cheatable state which has a list of cheaters, and, only then, I pass it here.

Reply:
I think the only thing you need to instantiate a cheater is cheatable state, so we don't really have to rely on passing them around. I'd be careful though, since we really don't want to interlink their logic, it will create spaghetti

Reply:
In this case unregistering should happen in handling the cheatable syscall, right? No need to unregister in the contracts cheater itself

Reply:
There are two scenarios when we unregister:
- `stop_expect` is called
- end of the scope is reached

So I think we kind of need this in both, a call's cheater and in the syscall handler